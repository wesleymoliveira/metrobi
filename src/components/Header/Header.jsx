import { Typography } from '../Typography/Typography';
import * as S from './Header.styled';

export function Header() {
	return (
		<S.Wrapper data-testid='header'>
			<Typography size='xxlarge'>
				Header
			</Typography>
		</S.Wrapper>
	);
}
