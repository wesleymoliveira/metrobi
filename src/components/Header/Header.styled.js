import styled, { css } from 'styled-components'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    background-color: ${theme.colors.headerBackground};
    height: 8rem;
    display: flex;
    justify-content: center;
    align-items: center;
  `}
`;
