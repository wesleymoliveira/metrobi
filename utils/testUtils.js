import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
