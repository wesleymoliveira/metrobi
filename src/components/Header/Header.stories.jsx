import React from 'react';
import { Header } from './Header';

const HeaderArgTypes = {
  label: {
    name: 'label',
    type: { name: 'string', required: false },
    description: 'overwritten description',
    table: {
      type: { 
        summary: 'something short', 
        detail: 'something really really long' 
      },
    },
    control: {
      type: 'string'
    }
  }
};

export default {
	title: 'components/Header',
	component: Header,
	argTypes: HeaderArgTypes
};

export const Default = (args) => <Header {...args} />;
