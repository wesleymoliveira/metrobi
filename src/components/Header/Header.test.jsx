import { screen } from '@testing-library/react';

import { Header } from './Header';
import { render, snapshotRenderer } from '../../../utils/testUtils';

describe('<Header />', () => {
		it('should render the component', () => {
		render(<Header />);
		
		expect(screen.getByTestId(/header/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<Header />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should match the original background-color', () => {
		const tree = snapshotRenderer(<Header />).toJSON();
		//make sure the background-color is the original one
		expect(tree).toHaveStyleRule('background-color', '#54d5eb')
	});

});
