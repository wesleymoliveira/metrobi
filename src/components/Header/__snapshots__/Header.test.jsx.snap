// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
.c1 {
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c0 {
  background-color: #54d5eb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<header
  className="c0"
  data-testid="header"
>
  <h1
    className="c1"
    color="white"
    data-testid="typography"
    size="xxlarge"
  >
    Header
  </h1>
</header>
`;
